src/material-experimental/theming/_checkbox.scss

```diff
@@ -0,0 +1,99 @@
+    )
+  );
+}
+
+@function _get-tokens-for-theme-type($type) {
+  $is-dark: $type == dark;
+  $foreground: if($is-dark, white, black);
+  $disabled-color: color.change($foreground, $alpha: 0.38);
+  $border-color: color.change($foreground, $alpha: 0.54);
+  $active-border-color: mat.get-color-from-palette(mat.$gray-palette, if($is-dark, 200, 900));
+
+  @return (
+    (mdc, checkbox): (
+      disabled-selected-icon-color: $disabled-color,
+      disabled-unselected-icon-color: $disabled-color,
+      unselected-focus-icon-color: $active-border-color,
+      unselected-hover-icon-color: $active-border-color,
+      unselected-icon-color: $border-color,
+      unselected-pressed-icon-color: $border-color,
+      unselected-focus-state-layer-color: $foreground,
+      unselected-hover-state-layer-color: $foreground,
+      unselected-pressed-state-layer-color: $foreground,
+    )
+  );
+}
+
+$_customization-defs: map.merge(
+  theming.rename-tokens((
+    checkmark-color: selected-checkmark-color,
+    disabled-checkmark-color: disabled-selected-checkmark-color,
+    selected-focus-ring-opacity: selected-focus-state-layer-opacity,
+    selected-hover-ring-opacity: selected-hover-state-layer-opacity,
+    selected-pressed-ring-opacity: selected-pressed-state-layer-opacity,
+    unselected-focus-ring-opacity: unselected-focus-state-layer-opacity,
+    unselected-hover-ring-opacity: unselected-hover-state-layer-opacity,
+    unselected-pressed-ring-opacity: unselected-pressed-state-layer-opacity,
+    disabled-selected-box-color: disabled-selected-icon-color,
+    disabled-unselected-box-color: disabled-unselected-icon-color,
+    selected-focus-box-color: selected-focus-icon-color,
+    selected-hover-box-color: selected-hover-icon-color,
+    selected-icon-color: selected-box-color,
+    selected-pressed-box-color: selected-pressed-icon-color,
+    unselected-focus-box-color: unselected-focus-icon-color,
+    unselected-hover-box-color: unselected-hover-icon-color,
+    unselected-box-color: unselected-icon-color,
+    unselected-pressed-box-color: unselected-pressed-icon-color,
+    selected-focus-ring-color: selected-focus-state-layer-color,
+    selected-hover-ring-color: selected-hover-state-layer-color,
+    selected-pressed-ring-color: selected-pressed-state-layer-color,
+    unselected-focus-ring-color: unselected-focus-state-layer-color,
+    unselected-hover-ring-color: unselected-hover-state-layer-color,
+    unselected-pressed-ring-color: unselected-pressed-state-layer-color,
+    ripple-size: state-layer-size,
+  ), (mdc, checkbox)),
+  (
+    color-palette: meta.get-function(_get-tokens-for-color-palette),
+    theme-type: meta.get-function(_get-tokens-for-theme-type),
+  ));
+
+// Configure the mat-checkbox's theme.
+@function checkbox($customizations: ()) {
+  @return (
+    id: 'mat.checkbox',
+    customizations:
+        theming.get-customized-tokens('mat.checkbox', $_customization-defs, $customizations),
+    deps: (),
+  );
```

Comment:
Also for later: it occurs to me that we should have a test suite for all of the components that runs their theming function and asserts that it has all of the correct fields.

Reply:
We do have something like this at the token level (it verifies that our token mapping includes all the tokens MDC supports), but we could consider changing it to check the actual emitted vars - that feels more e2e